source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -205,22 +211,31 @@ bool FaultFilter::isAbortEnabled() {
   return enabled;
 }
 
-absl::optional<uint64_t> FaultFilter::delayDuration() {
-  absl::optional<uint64_t> ret;
+absl::optional<std::chrono::milliseconds>
+FaultFilter::delayDuration(const Http::HeaderMap& request_headers) {
+  absl::optional<std::chrono::milliseconds> ret;
 
   if (!isDelayEnabled()) {
     return ret;
   }
 
-  uint64_t duration = config_->runtime().snapshot().getInteger(RuntimeKeys::get().DelayDurationKey,
-                                                               fault_settings_->delayDuration());
+  // See if the delay provider has a default delay, if not there is no delay.
```

Comment:
Update comment?  I'd say per-header delay doesn't have  default delay

Reply:
Also while this appears to be consistent with other fault filter runtime overrides, it might be worth clarifying docs that you can't simply turn up delay (or other features) via runtime override - they must be configured on the command line and then adjusted via runtime.